### PR TITLE
[12.0][IMP] currency_rate_update: allow instance access

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

File: currency_rate_update/models/res_currency_rate_provider.py [hunk 1]
```diff
@@ -285,11 +285,16 @@ def _scheduled_update(self):
 
         _logger.info('Scheduled currency rates update complete.')
 
-    @api.model
-    def _get_supported_currencies(self, service):
+    @api.multi
+    def _get_supported_currencies(self):  # pragma: no cover
+        self.ensure_one()
+
```
Comment (pedrobaeza) [inline currency_rate_update/models/res_currency_rate_provider.py]: Why adding this ugly empty lines inside a method?

File: currency_rate_update/models/res_currency_rate_provider.py [hunk 2]
```diff
@@ -285,11 +285,16 @@ def _scheduled_update(self):
...
 
-    @api.model
-    def _obtain_rates(self, service, base_currency, currencies, date_from,
-                      date_to):
+    @api.multi
+    def _obtain_rates(
+        self, base_currency, currencies, date_from, date_to
+    ):  # pragma: no cover
+        self.ensure_one()
+
```
Comment (pedrobaeza) [inline currency_rate_update/models/res_currency_rate_provider.py]: The same about the empty line

File: currency_rate_update/models/res_currency_rate_provider_ECB.py [hunk 3]
```diff
@@ -19,9 +19,11 @@ class ResCurrencyRateProviderECB(models.Model):
         selection_add=[('ECB', 'European Central Bank')],
     )
 
-    @api.model
-    def _get_supported_currencies(self, service):
-        if service == 'ECB':
+    @api.multi
+    def _get_supported_currencies(self):
+        self.ensure_one()
```
Comment (pedrobaeza) [inline currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: No need to repeat the `self.ensure_one()`. Even more. This method doesn't need to ensure that.

File: currency_rate_update/models/res_currency_rate_provider_ECB.py [hunk 4]
```diff
@@ -19,9 +19,11 @@ class ResCurrencyRateProviderECB(models.Model):
...
+
+        if self.service == 'ECB':
```
Comment (pedrobaeza) [inline currency_rate_update/models/res_currency_rate_provider_ECB.py]: It's better for indentation to put `if self.service != 'ECB': return super...` and develop the rest of the lines with no indentation.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: currency_rate_update/models/res_currency_rate_provider_ECB.py (hunk 3 above)
Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: it's not repeating, super's `self.esnure_one()` may not be called

File: currency_rate_update/models/res_currency_rate_provider.py (hunk 2 above)
Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider.py]: updated

File: currency_rate_update/models/res_currency_rate_provider.py (hunk 1 above)
Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider.py]: updated

File: currency_rate_update/models/res_currency_rate_provider_ECB.py (hunk 3 above)
Comment (pedrobaeza) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: OK, this is more a developer highlight than other thing, so it's not so important to be removed or not.

File: currency_rate_update/models/res_currency_rate_provider_ECB.py (hunk 4 above)
Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py]: Done

File: currency_rate_update/models/res_currency_rate_provider_ECB.py [hunk 5]
```diff
@@ -19,47 +19,51 @@ class ResCurrencyRateProviderECB(models.Model):
...
-                xml.sax.parse(response, handler)
-
-            return handler.content
-
-        return super()._obtain_rates(service, base_currency, currencies,
-                                     date_from, date_to)
+    @api.multi
+    def _get_supported_currencies(self):
+        self.ensure_one()
+
```
Comment (pedrobaeza) [inline currency_rate_update/models/res_currency_rate_provider_ECB.py]: Still putting useless empty lines inside a method...

Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py]: done

File: currency_rate_update/models/res_currency_rate_provider_ECB.py (hunk 3 above)
Comment (pedrobaeza) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: But I see more cases. I have put this one, but I mean all the empty lines inside a method.

Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: Well, logical separation?

Comment (pedrobaeza) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: We have talked about this other times:

* PEP8 encourages for not doing it.
* If you need a visual separation, better to put a comment line that also explains what the logical block is.
* This bloats the real number of lines/diff, and so on.

Comment (alexey-pelykh) [reply currency_rate_update/models/res_currency_rate_provider_ECB.py:24]: But code like
```
a = something()
# here goes comment
b = something()
```
is less readable than
```
a = something()

# here goes comment
b = something()
```

If that's so strict, why it's not enforced via lint?

**Comment by alexey-pelykh** [comment]
@pedrobaeza added comments and removed all empty lines that were not very useful to my understanding

**Review by pedrobaeza — APPROVED** [review]
Thanks for following my comments

**Review by tbaden — APPROVED** [review]
code review: lgtm

**Pull request merged**